src/Microsoft.Restier.Core/RestierContainerBuilder.cs

```diff
@@ -112,11 +115,8 @@ public IContainerBuilder AddService(ODataServiceLifetime lifetime, Type serviceT
         /// <returns>The container built by this builder.</returns>
         public virtual IServiceProvider BuildContainer()
         {
-            foreach (var descriptor in RestierServices)
-            {
-                Services.Add(descriptor);
-            }
-            AddRestierService(!RestierServices.Any());
+            postOdataConfigureAction?.Invoke(Services);
+            AddRestierService(!Services.Any(x => x.ServiceType == typeof(ApiBase)));
```

Comment:
This kind of changes the check, but I think adding the RestierService when no apiBase services have been registered on the services collection before is a sensible one.